### ISSUE TITLE
How to debug an error?

### ISSUE THREAD

**Issue by halo** [EugeneDae/vscode-refresh-browser#10]
After a couple hours of regular work, this extension stops to refresh the browser. Instead it shows this message:

<img width="382" alt="Screen Shot 2021-09-07 at 19 20 39" src="https://user-images.githubusercontent.com/11441/132385653-9388522c-3ab1-4997-ab7b-73b16e01daf4.png">

I have no idea what causes it. I already tried deactivating pretty much every other extension but the error keeps recurring after some time.

One time I thought that the problem starts happening when I save a file (CMD + s) and then quickly try to refresh the browser. But I cannot reproduce the error.

Is there any log file that could hint me as to what is going wrong?

Thank you for your time and for making this extension, it's very useful.

**Comment by EugeneDae** [comment]
> Is there any log file that could hint me as to what is going wrong?

Yes, please try:
![Screen Shot 2021-09-08 at 01 25 24](https://user-images.githubusercontent.com/980835/132418381-a9ace349-7741-40c3-9f7e-ec95ca03ceff.png)

Then switch to Console:

![Screen Shot 2021-09-08 at 01 25 46](https://user-images.githubusercontent.com/980835/132418402-a7abc66c-f4a5-452d-8e84-18fc00092565.png)

Thanks!

**Comment by EugeneDae** [comment]
Another thing to try to run the AppleScript code without the extension.

1. Using Spotlight, find Script Editor.app

2. Paste the code from below.

3. Script > Run.

4. See if it works and if there are any errors or warnings.

```
set URL_PREFIXES to {"https://localhost"}

tell application "Google Chrome"
	set _wIndex to 0
	set _tIndex to 0
	set _target to active tab of front window
	
	if ((count of URL_PREFIXES) > 0) then
		repeat with _window in windows
			set _wIndex to _wIndex + 1
			repeat with _tab in tabs of _window
				set _tIndex to _tIndex + 1
				
				if (my startswith(URL of _tab, URL_PREFIXES)) then
					set _target to _tab
					set active tab index of window _wIndex to _tIndex
					exit repeat
				end if
			end repeat
		end repeat
	end if
	
	reload _target
end tell

on startswith(_str, _list)
	repeat with i in _list
		if _str starts with i then return true
	end repeat
	false
end startswith
```

**Comment by halo** [comment]
Ok, thanks a lot for leading the way :)

It's the Apple Script failing when trying to access `front window` of `Google Chrome`

<img width="546" alt="Screen Shot 2021-09-08 at 08 51 04" src="https://user-images.githubusercontent.com/11441/132460457-a35ae9c1-dbd0-4818-9944-e2fcaa77a923.png">

I tried to debug it but. Apple Script says Chrome is running and that it is active, but it simply has no windows.

<img width="609" alt="Screen Shot 2021-09-08 at 08 56 01" src="https://user-images.githubusercontent.com/11441/132461221-0086419c-e19d-4db8-a59e-bc175cbbd76a.png">

 I also tried to [wait for it](https://stackoverflow.com/questions/66634858/getting-active-window-via-applescript-throws-error-can-t-get-window-1-of-proce) but the `window` would never exist:

<img width="541" alt="Screen Shot 2021-09-08 at 08 45 06" src="https://user-images.githubusercontent.com/11441/132461313-34b48a96-35f8-404f-82a5-8cfb6f4fe85c.png">

So I thought that the Google Chrome process must have run an automatic update of itself and maybe the OS can't access it any more. I quit Chrome - and I could not restart Chrome. I had to manually kill every helper and renderer process that was related to Chrome. Then I started Chrome again, and the script works again:

<img width="574" alt="Screen Shot 2021-09-08 at 08 56 05" src="https://user-images.githubusercontent.com/11441/132461563-31b2f412-c29b-4f1c-b590-627d2a9e63a7.png">

So – at least I know now that I don't need to restart my computer but that I have a work-around 😆 

It would be great, of course, if your extension could display the error message in a more prominent way, and maybe even explain what could have happened.

I almost just deleted the extension because I felt it was buggy. I'm glad I reached out to you, but it would be a shame if other people just delete it and think of it no more.

Again, thank you for your time and effort. I appreciate it.

**Comment by EugeneDae** [comment]
Nice investigation! So, it's a Chrome's quirk.

>It would be great, of course, if your extension could display the error message in a more prominent way, and maybe even explain what could have happened.

Done & published in version 1.1.3.

Thanks!

**Comment by halo** [comment]
Sweet! Thank you! 

I even got the pleasure to try it right away, because Chrome let me down again 😄 and it works:

<img width="379" alt="Screen Shot 2021-09-08 at 21 14 06" src="https://user-images.githubusercontent.com/11441/132570832-6a9207d2-dd03-423e-a14f-785a28fed5b4.png">
